main.py

```diff
@@ -31,12 +31,14 @@
 
 """
 
-from typing import Callable, Dict, Pattern, Optional, Union
+from typing import Dict, List, Pattern, Optional, Union
 
 import sys
-import os
 import re
 import logging
+from os import environ as ENV
+from platform import system as os_name
+from pathlib import Path
```

Comment:
I approve! Pathlib was introduced in Python 3.4 and is a much cleaner way of working with paths. Agree that the os.path stuff is ugly and hacky.